### PR TITLE
NETOBSERV-1764 frontend code styling followup

### PR DESCRIPTION
**Pull request by jpinsonneau**
## Description

Split netflow-traffic code that became too big for a single file
- netflow traffic model has been introduced to manage states
- drawer is now on its own separated component
- overview / table & topology are now under `tabs` folder with an additionnal tabs container component
- modals are centralized in modals component

## Dependencies

https://github.com/netobserv/network-observability-console-plugin/pull/541

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [X] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [X] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

**Comment by jpinsonneau** [comment]
@Amoghrd I rebased that PR and we should be good to go for a non regression testing :smile_cat: 

I had to merge https://github.com/netobserv/network-observability-console-plugin/pull/541 since we had too many changes impacting the codebase. Let's validate this one properly ! Thanks

**Comment by jpinsonneau** [comment]
Fixed column issue reported in jira in https://github.com/netobserv/network-observability-console-plugin/pull/551/commits/7e3cc4dac9f0aa46d7b0578bdbc0a14d2d7fce6e

**Comment by Amoghrd** [comment]
Reran the tests and all tests are now passing
/label qe-approved

**Comment by jpinsonneau** [comment]
Rebased without changes

@jotak I hope I didn't break your changes here :angel: If you can have a quick check I would appreciate

File: web/src/components/drawer/drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,406 @@
+import { K8sModel } from '@openshift-console/dynamic-plugin-sdk';
```
Comment (jotak) [inline web/src/components/drawer/drawer.tsx:1]: nit: can we name the file "netflow-traffic-drawer.tsx" so it matches the component name?

**Comment by jotak** [comment]
> Rebased without changes
> 
> @jotak I hope I didn't break your changes here 👼 If you can have a quick check I would appreciate

Tested and I didn't find any regression

**Comment by jotak** [comment]
Ideally, as part of this refactoring, it would be nice to move all the fetching functions in their respective sub-component (NetflowOverview, NetflowTable ...) ; so NetflowTraffic would only be responsible for doing the "switch(selectedView)", leaving the other components specialized in their domain, responsible both for fetching and displaying data. Currently, even with this refactoring, NetflowTraffic is still centralizing and dispatching everything, this is what I was hopeful we would be able to break

File: web/src/components/drawer/drawer.tsx (hunk 1 above)
Comment (jpinsonneau) [reply web/src/components/drawer/drawer.tsx]: done in https://github.com/netobserv/network-observability-console-plugin/pull/551/commits/1a3e846440b784fc3a4aa8fdb3e0518a03887b88

**Comment by jpinsonneau** [comment]
> Ideally, as part of this refactoring, it would be nice to move all the fetching functions in their respective sub-component (NetflowOverview, NetflowTable ...) ; so NetflowTraffic would only be responsible for doing the "switch(selectedView)", leaving the other components specialized in their domain, responsible both for fetching and displaying data. Currently, even with this refactoring, NetflowTraffic is still centralizing and dispatching everything, this is what I was hopeful we would be able to break

Yes that's something we can do using the `React.forwardRef` as same as in the `SearcComponent`.
The child view ref will be propagated to the parent and callable. Let's merge this PR first and do this in a followup PR.

**Pull request merged**